client/src/api/types/post.ts

```diff
@@ -24,6 +25,7 @@ export type GetPostsDto = {
 
 export type CreatePostReqDto = Pick<BasePostDto, 'title' | 'description'> & {
   tagname: string[]
+  topicname: string
```

Comment:
```suggestion
  topicname: string
```
- Given that topics are optional, we should allow for null topics to be specified
- I've sorely regretted being unable to intervene with casing of parameters back when others were handling the codebase. Could we use camelCasing here please?